Matplotlib deprecation warning in visualization docs.

### Describe the bug

```
___________________________________________________________________________ [doctest] visualization.rst ____________________________________________________________________________
069 By default and image animation is returned.
070 (See below to learn how to use ``plot_axes`` kwarg to produce a line animation.)
071 
072 For for data with two array axes, an image is produced similar to that of `matplotlib.pyplot.imshow`.
073 
074 .. plot::
075   :context:
076   :include-source:
077 
078   >>> ax = my_cube[0].plot()
UNEXPECTED EXCEPTION: MatplotlibDeprecationWarning('Auto-removal of overlapping axes is deprecated since 3.6 and will be removed two minor releases later; explicitly call ax.remove() as needed.')
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.6/x64/lib/python3.10/doctest.py", line 1350, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest visualization.rst[22]>", line 1, in <module>
  File "/home/runner/work/ndcube/ndcube/.tox/py310/lib/python3.10/site-packages/ndcube/ndcube.py", line 794, in plot
    return self.plotter.plot(*args, **kwargs)
  File "/home/runner/work/ndcube/ndcube/.tox/py310/lib/python3.10/site-packages/ndcube/visualization/mpl_plotter.py", line 84, in plot
    ax = self._plot_2D_cube(plot_wcs, axes, plot_axes, axes_coordinates,
  File "/home/runner/work/ndcube/ndcube/.tox/py310/lib/python3.10/site-packages/ndcube/visualization/mpl_plotter.py", line 157, in _plot_2D_cube
    axes = plt.subplot(projection=wcs, slices=plot_axes)
  File "/home/runner/work/ndcube/ndcube/.tox/py310/lib/python3.10/site-packages/matplotlib/pyplot.py", line 1274, in subplot
    _api.warn_deprecated(
  File "/home/runner/work/ndcube/ndcube/.tox/py310/lib/python3.10/site-packages/matplotlib/_api/deprecation.py", line 96, in warn_deprecated
    warn_external(warning, category=MatplotlibDeprecationWarning)
  File "/home/runner/work/ndcube/ndcube/.tox/py310/lib/python3.10/site-packages/matplotlib/_api/__init__.py", line 363, in warn_external
    warnings.warn(message, category, stacklevel)
matplotlib._api.deprecation.MatplotlibDeprecationWarning: Auto-removal of overlapping axes is deprecated since 3.6 and will be removed two minor releases later; explicitly call ax.remove() as needed.
/home/runner/work/ndcube/ndcube/docs/visualization.rst:78: UnexpectedException

---------- coverage: platform linux, python 3.10.6-final-0 -----------
Coverage XML written to file /home/runner/work/ndcube/ndcube/coverage.xml

============================================================================= short test summary info ==============================================================================
SKIPPED [1] ../../.tox/py310/lib/python3.10/site-packages/pytest_doctestplus/plugin.py:253: unable to import module PosixPath('/home/runner/work/ndcube/ndcube/.tox/py310/lib/python3.10/site-packages/ndcube/utils/sphinx/code_context.py')
SKIPPED [4] ../../.tox/py310/lib/python3.10/site-packages/_pytest/doctest.py:452: all tests skipped by +SKIP option
XFAIL ../../.tox/py310/lib/python3.10/site-packages/ndcube/extra_coords/tests/test_extra_coords.py::test_extra_coords_2d_quantity
  >1D Tables not supported
XFAIL ../../.tox/py310/lib/python3.10/site-packages/ndcube/extra_coords/tests/test_lookup_table_coord.py::test_2d_nout_1_no_mesh
  >1D Tables not supported
XFAIL ../../.tox/py310/lib/python3.10/site-packages/ndcube/extra_coords/tests/test_lookup_table_coord.py::test_2d_skycoord_no_mesh
  >1D Tables not supported
XFAIL ../../.tox/py310/lib/python3.10/site-packages/ndcube/extra_coords/tests/test_lookup_table_coord.py::test_2d_quantity
  >1D Tables not supported
XFAIL ../../.tox/py310/lib/python3.10/site-packages/ndcube/extra_coords/tests/test_lookup_table_coord.py::test_slicing_quantity_table_coordinate_2d
  >1D Tables not supported
XFAIL ../../.tox/py310/lib/python3.10/site-packages/ndcube/extra_coords/tests/test_lookup_table_coord.py::test_2d_nout_1_no_mesh_slice
  >1D Tables not supported
XFAIL ../../.tox/py310/lib/python3.10/site-packages/ndcube/extra_coords/tests/test_lookup_table_coord.py::test_mtc_dropped_quantity_table
  >1D Tables not supported
XFAIL ../../.tox/py310/lib/python3.10/site-packages/ndcube/extra_coords/tests/test_lookup_table_coord.py::test_mtc_dropped_quantity_inside_table_no_mesh
  >1D Tables not supported
XFAIL ../../.tox/py310/lib/python3.10/site-packages/ndcube/tests/test_ndcube.py::test_axis_world_coords_complex_ec
  >1D Tables not supported
XPASS ../../.tox/py310/lib/python3.10/site-packages/ndcube/extra_coords/tests/test_lookup_table_coord.py::test_2d_skycoord_mesh_slice >1D Tables not supported
XPASS ../../.tox/py310/lib/python3.10/site-packages/ndcube/extra_coords/tests/test_lookup_table_coord.py::test_2d_skycoord_no_mesh_slice >1D Tables not supported
FAILED ../../docs/visualization.rst::visualization.rst
```

### To Reproduce

import sunpy

sunpy.map.Map(...)
etc


### Screenshots

_No response_

### System Details

_No response_

### Installation method

_No response_

--- a/docs/visualization.rst
+++ b/docs/visualization.rst
@@ -58,6 +58,11 @@ The plot method can be called very simply.
   >>> ax = my_cube.plot()
   >>> plt.show()
 
+.. plot::
+  :nofigs:
+
+  >>> plt.clf()
+
 Note how no arguments are required.
 The necessary information to generate the plot is derived from the data and metadata in the `~ndcube.NDCube`.
 The axis labels are taken from the WCS axis names defined in ``my_cube.wcs.wcs.cname``.
@@ -78,6 +83,11 @@ For for data with two array axes, an image is produced similar to that of `matpl
   >>> ax = my_cube[0].plot()
   >>> plt.show()
 
+.. plot::
+  :nofigs:
+
+  >>> plt.clf()
+
 For data with one array axis, a line plot is produced, similar to `matplotlib.pyplot.plot`.
 
 .. plot::
@@ -86,6 +96,11 @@ For data with one array axis, a line plot is produced, similar to `matplotlib.py
 
   >>> ax = my_cube[1, 1].plot()
   >>> plt.show()
+
+.. plot::
+  :nofigs:
+
+  >>> plt.clf()
 
 Setting the x and y ranges of the plot can be done simply by indexing the `~ndcube.NDCube` object to the desired region of interest and then calling the plot method, e.g.
 

